### PR TITLE
Add localization pack for Armenia

### LINKED ISSUE #18079: Add localization pack for Armenia
#### Describe the solution you'd like

I have added a localization pack for Armenia, please include it in future versions.

#### Additional context

Thank you very much for such hard work :)

### PR DESCRIPTION
**Pull request by mrAKAR**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add localization pack for Armenia
| Type?         | new feature 
| Category?     | LO
| BC breaks?    | no
| Deprecations? |  no
| Fixed ticket? | Fixes #18079 
| How to test? | * Go in International > Localization<br>* Install AM (Armenia) localization pack<br>* Go in Languages tab & Check the Armenian language is installed<br>* Go in Currencies tab & Check the Armenian Dram is installed<br>* Go on FrontOffice & Check the currency is well displayed

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18070)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

File: localization/am.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<localizationPack name="Armenian" version="1.0">
```
Comment (LouiseBonnard) [inline localization/am.xml]:
```suggestion
<localizationPack name="Armenia" version="1.0">
```

File: localization/am.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	<currencies>
+		<currency name="Armenian Dram" iso_code="AMD" iso_code_num="051" sign="֏" blank="1" conversion_rate="0.0021" format="2" decimals="0" />
```
Comment (LouiseBonnard) [inline localization/am.xml]:
```suggestion
		<currency name="Armenian dram" iso_code="AMD" iso_code_num="051" sign="֏" blank="1" conversion_rate="0.0021" format="2" decimals="0" />
```

File: localization/am.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    
+		<taxRulesGroup name="AM Standard Rate (20%)">
+			<taxRule iso_code_country="am" id_tax="1" />
+		</taxRulesGroup>
+
+		<taxRulesGroup name="AM Reduced Rate (16.67%)">
+			<taxRule iso_code_country="am" id_tax="2" />
+		</taxRulesGroup>
+    
+		<taxRulesGroup name="AM Reduced Rate (0%)">
```
Comment (LouiseBonnard) [inline localization/am.xml]:
```suggestion
		<taxRulesGroup name="AM Zero Rate (0%)">
```

File: localization/am.xml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+<localizationPack name="Armenian" version="1.0">
+	<currencies>
+		<currency name="Armenian Dram" iso_code="AMD" iso_code_num="051" sign="֏" blank="1" conversion_rate="0.0021" format="2" decimals="0" />
+	</currencies>
+	<languages>
+		<language iso_code="hy" />
+	</languages>
+	<taxes>
+		<tax id="1" name="ԱԱՀ AM 20%" rate="20" />
+		<tax id="2" name="ԱԱՀ AM 16.67%" rate="16.67" />
```
Comment (LouiseBonnard) [inline localization/am.xml]: Isn't this rate the same as the previous 20%? I can read [here](http://www.ardani.net/armenia/armenia-taxation-vad.php): 'The general VAT rate is 20% of the turnover of taxable goods and services, which is equal to 16,67% of VAT-inclusive prices'.

**Comment by LouiseBonnard** [comment]
@mrAKAR, thanks for your contribution here, it's very nice! I've left a few comments, I let you address them and stay available if you need further information. :-)

PS: each pull request must be linked to an issue, could you please open an issue and link it to this PR so that we can include your contribution in future versions of PrestaShop?

File: localization/am.xml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+	<taxes>
+		<tax id="1" name="ԱԱՀ AM 20%" rate="20" />
+		<tax id="2" name="ԱԱՀ AM 0%" rate="0" />
+    
+		<taxRulesGroup name="AM Standard Rate (20%)">
+			<taxRule iso_code_country="am" id_tax="1" />
+		</taxRulesGroup>
+    
+		<taxRulesGroup name="AM Zero Rate (0%)">
+			<taxRule iso_code_country="am" id_tax="3" />
```
Comment (LouiseBonnard) [inline localization/am.xml]:
```suggestion
			<taxRule iso_code_country="am" id_tax="2" />
```

**Comment by LouiseBonnard** [comment]
@mrAKAR, thanks a lot for the code modifications! Here is the link to create the associated issue: https://github.com/PrestaShop/PrestaShop/issues/new?template=feature_request.md

**Review by LouiseBonnard — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]
thanks @mrAKAR !

@LouiseBonnard do you think this should be tested or merged directly?

**Comment by LouiseBonnard** [comment]
@matthieu-rolland, it should (always) be tested. :-)

**Comment by Progi1984** [comment]
> @matthieu-rolland, it should (always) be tested. :-)

Good answer @LouiseBonnard ;)

**Review by matks — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @mrAKAR,

First:
I make an install with the "English" language and "Armenia" country. => OK
In the FO, the currency symbol is not well displayed
image
In the BO => ok
image

Second:
In the Install page "Choose your language", the `Հայերէն (Armenian)` language doesn't exist.
image


Thanks to check and feedback.

**Comment by LouiseBonnard** [comment]
Hey @khouloudbelguith, localization packs apply to a store when it is already installed. All languages in the back office are not available in the installer. Languages available in the installer are listed [here](https://github.com/PrestaShop/PrestaShop/tree/develop/install-dev/langs). :-)

**Comment by khouloudbelguith** [comment]
Hi @LouiseBonnard,

Thanks for these clarifications :slightly_smiling_face:

**Comment by colinegin** [comment]
Hello @LouiseBonnard what is the status of this issue ?

**Comment by LouiseBonnard** [comment]
It has been tested by @khouloudbelguith and it appears that the currency symbol is not properly displayed. But I have no clue how to fix that. I guess it should be handled by the CLDR.

**Comment by colinegin** [comment]
ping @PrestaShop/prestashop-core-developers what do you think of this problem ?

**Comment by LouiseBonnard** [comment]
Hi @khouloudbelguith, it might be the theme's font that does not support this type of characters, to be tested with another font to see if the problem comes from the used font.

**Comment by Progi1984** [comment]
@LouiseBonnard I confirm. Open Sans doesn't support this glyph `֏` :

image

**Comment by LouiseBonnard** [comment]
Hi @Progi1984, thanks a lot for confirming! @SimonGrn, I guess it should be tested with another font then? Could I label this PR as _waiting for QA_?

**Comment by SimonGrn** [comment]
Hi @LouiseBonnard ,

We can test with another font, however that problem means it will be broken with every front theme that use this font (and probably 99% of themes deriving from Classic).
But there's nothing we can do to make this font displays correctly the symbol.
There is no elegant solution as far as I know.
@Progi1984 do you have any suggestion ? Replacing the symbol with the 3-letter name ?

File: localization/am.xml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<localizationPack name="Armenia" version="1.0">
+	<currencies>
+		<currency name="Armenian dram" iso_code="AMD" iso_code_num="051" sign="֏" blank="1" conversion_rate="0.0021" format="2" decimals="0" />
```
Comment (Progi1984) [inline localization/am.xml:4]:
```suggestion
		<currency name="Armenian dram" iso_code="AMD" iso_code_num="051" blank="1" conversion_rate="0.0021" format="2" decimals="0" />
```

The sign is not anymore used. The symbol comes from CLDR data.

**Review by Progi1984 — COMMENTED** [review]
New information.

**Comment by Progi1984** [comment]
Ok. I have more data. It is logical that the symbol `֏` is not fetched because it doesn't come from this xml. The symbol is fetched in CLDR data. For having the symbol, you need to have the shop in armenian (hy) as defaut language and add AMD currency ;) 

Ping @SimonGrn & @LouiseBonnard

File: localization/am.xml (hunk 6 above)
Comment (mrAKAR) [reply localization/am.xml:4]: May be will be better to replace the sign with "դրամ", it's in Armenian and means the same.

**Comment by SimonGrn** [comment]
So that means you can't display the symbol if you want to have the AMD currency in an italian shop for example ? You can only display it correctly in an armenian context ? That's kinda problematic ?

**Comment by Progi1984** [comment]
@SimonGrn We have fixed a bug.

Here's a test scenario :
* Install PrestaShop (in French in my test)
* Connect to BackOffice
* Install the localization pack for Armenia
* In FrontOffice, in French with Armenian Dram,
image
* In FrontOffice, in Armenian with Armenian Dram,
image

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi,

In fact, I tried to make install with `French` language & `Armenia` country.
When `French` is selected => The currency symbol is **ok** as confirmed by @Progi1984.
When `Հայերէն (Armenian)` language is selected, the symbol is **NOK**
BO
image
Fo
image
image

I make some search, I found this 
image

@SimonGrn @LouiseBonnard @Progi1984, so can we confirm this behavior?
Thanks!

**Comment by SD1982** [comment]
@LouiseBonnard WDYT ?

**Comment by LouiseBonnard** [comment]
Hi @SD1982, thanks for pinging me here. I think it is problematic that the symbol is not properly displayed when the FO language is Armenian. It seems more likely to me that most of the people who will browse a store with the AMD currency will do it in Armenian.

Is the symbol mandatory? As long as it is handled by the CLDR, perhaps a simple workaround would be to display only the currency ISO code, AMD, instead of the symbol?

**Comment by SD1982** [comment]
Ping @khouloudbelguith

**Comment by khouloudbelguith** [comment]
Hi @LouiseBonnard,

Thanks for your feedback.
>Is the symbol mandatory? 

Yeah, I think, it is => the symbol will be displayed in the FO, in the BO > Orders & Invoices
If it is not displayed, it is a problem, @LouiseBonnard what do you think?

> As long as it is handled by the CLDR, perhaps a simple workaround would be to display only the currency ISO code, AMD, instead of the symbol?

I think it is a good idea.

Thanks!

**Comment by LouiseBonnard** [comment]
@khouloudbelguith, I don't see many other options in this case, I think we should forget about the symbol since it cannot be displayed properly and rather display the currency ISO code, AMD.

**Comment by jolelievre** [comment]
I guess the symbol displayed or not depends on the environment and the Fonts installed on your computer
@Progi1984 in your case it worked correctly right?

So indeed probably not all computers will be able to display this (especially people from France ^^) but maybe most Armenian computers can see the symbol correctly no?

Also one question I have is how comes the symbol doesn't come from the CLDR? 🤔  @Progi1984 maybe you have more insight on this? Seems like in the CLDR the symbol is not defined: https://github.com/PrestaShop/PrestaShop/blob/develop/localization/CLDR/core/common/main/am.xml#L5338

**Comment by jolelievre** [comment]
Side note, I couldn't find the symbol in the latest version of CLDR either (37, we use 36)
https://github.com/unicode-org/cldr/blob/5ab4c86252c061559213460cc007667ea164bd3a/common/main/am.xml#L5453

However in the master branch there is something new:
https://github.com/unicode-org/cldr/blob/master/common/main/am.xml#L5458

I guess we will need to wait for version 38, last version is from April 2020, release cycles of CLDR doesn't seem to exceed 7months So maybe we'll be able to update to version 38 bu the end of our 1.7.8 version development 🤷‍♂️

**Comment by jolelievre** [comment]
Good news the CLDR is being updated https://github.com/PrestaShop/PrestaShop/pull/21680
This should unlock this PR ^^

**Comment by Progi1984** [comment]
@LouiseBonnard Good news. This PR is ready for QA. I improved the "How to test?" section.

**Comment by khouloudbelguith** [comment]
Hi @mrAKAR, @LouiseBonnard, @Progi1984,

I tried to make install with `English` language & `France` country.

0. Go in International > Localization & Install AM (Armenia) localization pack
1. In the **FO**, when `French` is selected => The currency symbol is NOK
image
2. In the **FO**, When `Հայերէն (Armenian)` language is selected, the symbol is NOK
image
3. In the BO > International > Localization > Currencies tab, the symbol of `Armenian Dram` currency is NOK
image
image


I tried to make install with `French` language & `Armenia` country.
1. In the **FO**, when `French` is selected => The currency symbol is NOK
image
2. In the **FO**, When `Հայերէն (Armenian)` language is selected, the symbol is NOK
image
3.  In the BO > International > Localization > Currencies tab, the symbol of `Armenian Dram` currency is NOK

image


Thanks!

**Comment by Progi1984** [comment]
@khouloudbelguith I really don't understand. I tried yesterday, and in Firefox, the currency was well displayed.

**Comment by SimonGrn** [comment]
Something to do with local cache ? Browser version or language ? Localization ?

**Comment by khouloudbelguith** [comment]
Hi @Progi1984, @SimonGrn,

I used Google chrome Version 85.0.4183.102 (Build officiel) (64 bits)
I installed a new fresh shop using an incognito browser.
tested with PHP version 7.2 & PHPv7.1

Thank you!

**Comment by Progi1984** [comment]
@khouloudbelguith @SimonGrn  It's weird.

Tested : `develop` branch and only the file `localization/am.xml`.

PHP 7.2.24 installed.

Install in Language : French / Country : Armenia.

Chrome : `Version 87.0.4280.88 (Build officiel) (64 bits)`

image
image
image

Firefox : `84.0 (64 bits)`
image
image
image

**Comment by Progi1984** [comment]
Idem for @NeOMakinG 
image

**Comment by khouloudbelguith** [comment]
Hi @Progi1984,

I check it again with firefox 45.6.0
https://drive.google.com/file/d/1S6fyok65cXSpnbM7dX-AhoUnXyQxFWXZ/view?usp=sharing
Same issue => could be a problem in my case, Ping @PrestaShop/qa-functional could you please check it.
Thanks!

**Comment by SD1982** [comment]
LGTM !!
image
image
image

Test 1
php 7.2 
installation
Language => french
country => Arménia
chrome => Version 87.0.4280.88 (Build officiel) (64 bits)

Test 2
installation
Language => english
country => fFrance
Add Armenia localisation pack
chrome => Version 87.0.4280.88 (Build officiel) (64 bits)

Thanks @mrAKAR !!

**Comment by Progi1984** [comment]
Thanks @mrAKAR & @SD1982 & @khouloudbelguith & @LouiseBonnard

**Pull request merged**